conformance/OWNERS

```diff
@@ -2,8 +2,8 @@
 # See the OWNERS_ALIASES file at https://github.com/kubernetes-sigs/gateway-api/blob/main/OWNERS_ALIASES for a list of members for each alias.
 
 approvers:
-  - sig-network-leads
-  - gateway-api-maintainers
```

Comment:
Does this remove maintainers from being approvers as well? Personally, I don't usually have bandwidth for much conformance review, so it's okay with me, but do we need escape hatches here?

Reply:
This was already covered by the higher level file, so felt like there was no need to repeat it here:

https://github.com/kubernetes-sigs/gateway-api/blob/a89369a8b66d29a04d493e27bece465811b204d7/OWNERS#L4-L6